src/main/java/org/broadinstitute/hellbender/utils/genotyper/UnfilledReadsLikelihoods.java

```diff
@@ -0,0 +1,123 @@
+package org.broadinstitute.hellbender.utils.genotyper;
+
+import htsjdk.samtools.util.Locatable;
+import htsjdk.variant.variantcontext.Allele;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import org.broadinstitute.hellbender.utils.IndexRange;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.pileup.PileupElement;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+//TODO comment this off
+public class UnfilledReadsLikelihoods<A extends Allele> extends ReadLikelihoods<A> {
+
+    private Map<String, List<PileupElement>> stratifiedPileups;
+
+    /**
+     * Constructs a new read-likelihood collection.
+     * <p>
+     * <p>
+     * The initial likelihoods for all allele-read combinations are
+     * 0.
+     * </p>
+     *
+     * @param samples all supported samples in the collection.
+     * @param alleles all supported alleles in the collection.
+     * @param reads   reads stratified per sample.
+     * @throws IllegalArgumentException if any of {@code allele}, {@code samples}
+     *                                  or {@code reads} is {@code null},
+     *                                  or if they contain null values.
+     */
+    public UnfilledReadsLikelihoods(SampleList samples, AlleleList<A> alleles, Map<String, List<GATKRead>> reads) {
+        super(samples, alleles, reads);
+    }
+
+    // Internally used constructor.
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private UnfilledReadsLikelihoods(final AlleleList alleles,
+                            final SampleList samples,
+                            final GATKRead[][] readsBySampleIndex,
+                            final Object2IntMap<GATKRead>[] readIndex,
+                            final double[][][] values) {
+       super(alleles, samples, readsBySampleIndex, readIndex, values);
+    }
+
+    /**
+     * Is this container expected to have the per-allele likelihoods calculations filled in.
+     */
+    public boolean hasFilledLikelihoods() {
+        return false;
+    }
+
+    /**
+     * Collect a map stratified per-sample of the base pileups at the provided Location
+     *
+     * @param loc reference location to construct pileups for
+     * @return
+     */
+    public Map<String, List<PileupElement>> getStratifiedPileups(final Locatable loc) {
+        if (stratifiedPileups != null) {
```

Comment:
This seems like an awkward duplication of the functionality from ReadsPileup.splitBySample(), but this is probably more efficient.  Maybe we can unify / speed that up?  Save for future maybe.